executor/executor.go

```diff
@@ -1332,7 +1332,10 @@ func (e *TableScanExec) nextChunk4InfoSchema(ctx context.Context, chk *chunk.Chu
 			columns[i] = table.ToColumn(colInfo)
 		}
 		mutableRow := chunk.MutRowFromTypes(retTypes(e))
-		err := e.t.IterRecords(e.ctx, nil, columns, func(_ kv.Handle, rec []types.Datum, cols []*table.Column) (bool, error) {
+		type tableIter interface {
+			IterRecords(sessionctx.Context, []*table.Column, table.RecordIterFunc) error
+		}
+		err := (e.t.(tableIter)).IterRecords(e.ctx, columns, func(_ kv.Handle, rec []types.Datum, cols []*table.Column) (bool, error) {
```

Comment:
Why do we need this interface?

Reply:
It's used by the `TableScanExec` here https://github.com/pingcap/tidb/blob/355fdfee4b19eed847ea9f4ab12c2acc9a772141/executor/builder.go#L1594

I guess it's a half-way refactoring for infoschema table reader, not all the tables there are rewritten using the new code...  @tangenta